src/components/Composer.vue

```diff
@@ -372,6 +393,37 @@ export default {
 		onAddCloudAttachment() {
 			this.bus.$emit('onAddCloudAttachment')
 		},
+		onSaveAndClose() {
+			this.state = STATES.SAVING
+			this.saveDraft(this.getMessageData)
+			wait(1500) // Illusion of things going on behind the scenes
+				.then(() => (this.state = STATES.SAVED))
+				.then(() => wait(1000)) // Time to register the "success"-message
+				.then(() => this.$router.push({name: 'home'})) // TODO: Navigate to better place
+		},
+		onDelete() {
+			this.state = STATES.DISCARDING
+			this.draftsPromise
+				.then(this.getMessageData)
+				.then(message => {
+					// TODO: Somehow delete the draft
+					// We've got `message`, but we need an `envelope`. I'm not 100%
```

Comment:
Essentially both represents a message. And in the old code base before the Vue rewrite we used just *message* for both objects that had the *from*, *to*, *cc*, *bcc*, *subject*, *flags*,... data, and the objects that contained the body data and attachments. Thus I renamed the former to gain some clarity in code.

I think we can just refactor the action to take *account id*, *folder id* and *message id* rather than an object. Then it will work with both envelopes and messages. Ironically, `deleteMessage` currently takes an `envelope` :see_no_evil:

Reply:
Ah, thank you for the clarification! 🙌 I'll look into a refactor and block this PR with another one. One potential problem I see is that I have neither `folderId` nor `messageId` at this stage, just the `draftUID` and `accountId`. I'll figure out if that's a problem when I start working on it 😅

>  Ironically, deleteMessage currently takes an envelope 🙈

Yeah, that tripped me up at first 😂